### PR TITLE
Format TS expressions when any script tag has `lang="ts"`

### PR DESCRIPTION
**Pull request by seiyab**
## Description
Fix https://github.com/prettier/prettier/issues/14575

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] ~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).~
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/format/vue/multiparser/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -70,6 +70,60 @@ export default {
...
+let x:      string |         number = 1
+</script>
+<script></script>
+<script></script>
+<script></script>
+<template>
+  <span
+  v-if=" (x    as string).length  >     0"
+  v-for="a in [1,2,  3,4,5].map(   (x  :    number) => x *   x)"
+  :foo="  (x    as number).toFixed(   2) "
```
Comment (seiyab) [inline tests/format/vue/multiparser/__snapshots__/jsfmt.spec.js.snap:95]: These aren't formatted in this branch due to https://github.com/prettier/prettier/issues/14432 .
Rebasing to next branch, they are formatted.
Feel free to ask me to
- remove these lines
- or, change base branch

**Review by sosukesuzuki — COMMENTED** [review]
Maybe this will conflicts with https://github.com/prettier/prettier/pull/14559. You may want to work on `next` as a base branch after #14559 has been merged. If we don't mind resolving conflicts, we can leave it as is.

@fisker What do you think?

**Comment by fisker** [comment]
It's fine to merge into main, #14559 already need update.

File: tests/format/vue/multiparser/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (fisker) [reply tests/format/vue/multiparser/__snapshots__/jsfmt.spec.js.snap:95]: We can ignore it.

File: src/language-html/print-preprocess.js [hunk 2]
```diff
@@ -412,14 +412,13 @@ function addIsSpaceSensitive(ast, options) {
...
       isVueScriptTag(child, options)
     );
-    if (!vueScriptTag) {
-      return;
-    }
-    const { lang } = vueScriptTag.attrMap;
-    if (lang === "ts" || lang === "typescript") {
+    const shouldMarkTs = vueScriptTags.some((vueScriptTag) =>
+      ["ts", "typescript"].includes(vueScriptTag.attrMap.lang)
+    );
```
Comment (fisker) [inline src/language-html/print-preprocess.js:420]: `filter` + `some` can reduce to one `.some`.

**Pull request merged**